README.rst

```diff
@@ -1,95 +1,167 @@
+decorators to guarantee inheritance "promises".
+
+Three inheritance patterns are provided:
+
+* implementing
+* overriding
+* augmenting
+
+Using the ``ipromise`` inheritance patterns can ensure an inheritance hierarchy
+is used as intended by forcing a build-time or run-time failure when not.
+
+----
+
+.. contents::
+
+----
+
+**********
+Installing
+**********
 
-* implementing,
-* overriding, and
-* augmenting.
+.. code-block:: shell
 
-Base class
-==========
-Checking promises depends on inheritance from the base class ``AbstractBaseClass``.  Unlike the standard library's similar class ``abc.ABCMeta``, ``AbstractBaseClass`` does not bring in any metaclasses.  This is thanks to Python 3.6's PEP 487, which added ``__init_subclass__``.
+    python -m pip install ipromise
+
+***
+Use
+***
+
+Base class ``AbstractBaseClass``
+================================
+
+Checking promises depends on inheritance from the base class
+``AbstractBaseClass``.  Unlike the standard library's similar class
+``abc.ABCMeta``, the ``AbstractBaseClass`` does not bring in any metaclasses.
+This is thanks to Python 3.6 `PEP 487
+<https://peps.python.org/pep-0487/>`_
+which added ``__init_subclass__``.
+
+Implementing with ``implements``
+================================
+
+*Implementing* is the pattern whereby an inheriting class's method implements an
+abstract method from a base class method.
 
-Implementing
-============
-*Implementing* is the pattern whereby an inheriting class's method implements an abstract method from a base class method.
 It is declared using the decorators:
 
-* ``abc.abstractmethod`` from the standard library, and
-* ``implements``, which indicates that a method implements an abstract method in a base class
+* ``abc.abstractmethod`` from the standard library
+* ``implements`` which indicates that a method implements an abstract method in
+  a base class
 
-For example:
+From ``samples/implements.py`` :
 
 .. code-block:: python
 
-    class HasAbstractMethod(AbstractBaseClass):
-        @abstractmethod
-        def f(self):
-            raise NotImplementedError
+    from ipromise import AbstractBaseClass, implements
+    import abc
 
+    class MyInterface(AbstractBaseClass):
+        @abc.abstractmethod
+        def f(self):
+            raise NotImplementedError("You forgot to implement f()")
 
-    class ImplementsAbstractMethod(HasAbstractMethod):
-        @implements(HasAbstractMethod)
+    class MyImplementation(MyInterface):
+        @implements(MyInterface)
         def f(self):
+            print("MyImplementation().f()")
             return 0
 
-Overriding
-==========
-*Overriding* is the pattern whereby an inheriting class's method replaces the implementation of a base class method.
-It is declared using the decorator ``overrides``, which marks the overriding method.
+    MyImplementation().f()
+
 
-An overriding method could call super, but does not have to:
+Overriding with ``overrides``
```

Comment:
I think these section headers are better without the added description "Overriding".  I think it's simpler.  Don't need to cram so much information into the headings.